CMakeLists.txt

```diff
@@ -60,10 +60,12 @@ set(FlatHash_SRCS
 )
 
 set(FlatBuffers_Tests_SRCS
+  # FLATBUFFERS_TRACK_VERIFIER_BUFFER_SIZE macro based compilation does not happen
```

Comment:
This macro has caused me a lot of grief :) and this change is actually not the right one now that I have looked into it a bit more. The core issue is that test.cpp and reflection.cpp both include flatbuffers.h (either directly or transitively) and test.cpp is defining the macro and reflection.cpp is not. So the end result is that both translation units see a different version of the header as they are getting compiled. The latest issue I saw while incorporating code review changes was stack corruption. The reason was that class Verifier is of different size in test.cpp and reflection.cpp. If I do sizeof(Verifier) in test.cpp its 28 bytes (32bit build) and 24 bytes in reflection.cpp. This reason is again this macro if I look inside the Verifier class then I see the same macro  
```
#ifdef FLATBUFFERS_TRACK_VERIFIER_BUFFER_SIZE
mutable const uint8_t *upper_bound_;
#endif
```

And this causes the difference. I can start looking into how we can propagate this macro to all translation units or maybe if you have some suggestion then please share that. I think cmake has something for this but I would have to look again.